src/asm/lexer.c

```diff
@@ -736,7 +736,9 @@ static char const *readMacroArg(char name)
 static int peekInternal(uint8_t distance)
 {
 	for (struct Expansion *exp = lexerState->expansions; exp; exp = exp->parent) {
-		assert(exp->offset < exp->size);
+		assert(exp->offset <= exp->size);
+		if (exp->offset == exp->size)
```

Comment:
Maybe add a comment explaiing in which case this gets triggered?
Additionally, this check is unnecessary, as if it's true, the rest of the loop is a no-op anyway.

Reply:
If you have `foo EQUS "bar + 2"`, `bar EQUS "qux"`, and `qux EQUS "hello"`, then to tokenize "`hello`" as an identifier the lexer has to `peek` past it. To do that, `peekInternal` has to `continue` down from the expansion of `qux`, then from the expansion of `bar`, into the expansion of `foo`, and get the space character from there.

I'll add a comment to that effect.

Reply:
If this check is true, then `exp->size - exp->offset` is zero, thus the check below is guaranteed to be false (since `distance` is unsigned, therefore positive), and then `distance` will be decreased by 0.
Hence, the rest of the loop is implicitly skipped anyway.

Reply:
(Since it's an exceptional case, I still think it's worth adding a comment, just not special-case it. I'm curious as to the compiler's behavior regarding this, though :P)

Reply:
Added a comment.